File: solutions/1256-rank-transform-of-an-array/solution.py
class Solution:
    def arrayRankTransform(self, arr: List[int]) -> List[int]:
        rank = {}
        sort_arr =sorted(list(set(arr)))

        for i in range(len(sort_arr)):
            rank[sort_arr[i]] = i+1

        for i in range(len(arr)):
            arr[i]=rank[arr[i]]

        return arr

File: solutions/1256-rank-transform-of-an-array/solution.java
class Solution {
    public int[] arrayRankTransform(int[] arr) {
        HashMap<Integer, Integer> numToRank = new HashMap<>();

        TreeSet<Integer> nums = new TreeSet<>();
        for(int i :arr){
            nums.add(i);
        }
        int rank=1;
        for(int i : nums){
            numToRank.put(i,rank);
            rank++;
        }
        for(int j=0;j<arr.length;j++){
            arr[j] = numToRank.get(arr[j]);
        }
        return arr;
            }
}
